tdmpc2/tdmpc2.py

```diff
@@ -279,12 +304,12 @@ def update(self, buffer):
 
 		# Return training statistics
 		self.model.eval()
-		return {
-			"consistency_loss": float(consistency_loss.mean().item()),
-			"reward_loss": float(reward_loss.mean().item()),
-			"value_loss": float(value_loss.mean().item()),
+		return TensorDict({
+			"consistency_loss": consistency_loss,
+			"reward_loss": reward_loss,
+			"value_loss": value_loss,
 			"pi_loss": pi_loss,
-			"total_loss": float(total_loss.mean().item()),
-			"grad_norm": float(grad_norm),
-			"pi_scale": float(self.scale.value),
-		}
+			"total_loss": total_loss,
+			"grad_norm": grad_norm,
+			"pi_scale": self.scale.value,
+		}).detach().mean()
```

Comment:
we need to return only detached tensors
I rely on tensordict's mean and detach for this